Parse/Internal/File/State/PFFileState.m

```diff
@@ -78,13 +78,11 @@ - (NSString *)secureURLString {
     }
 
     NSString *scheme = components.scheme;
-    if (![scheme isEqualToString:@"http"]) {
-        return self.urlString;
-    }
-
-    if ([components.host isEqualToString:_PFFileStateSecureDomain]) {
+    if ([scheme isEqualToString:@"http"] || [components.host isEqualToString:_PFFileStateSecureDomain]) {
```

Comment:
You are forcing everything here to use `https`, even though the url being used was `http`. The intention of this code is to use `https` if we can (either we are using a parse.com file domain or we aren't using `http` sent from the server.